src/service/resources-impl.js

```diff
@@ -1723,7 +1723,7 @@ export class Resource {
       return Promise.resolve();
     }
 
-    if (!this.renderOutsideViewport() && !this.isInViewport()) {
+    if (!this.isInViewport() && !this.renderOutsideViewport()) {
```

Comment:
We also need to discuss the confusion that will be developing between "can I render" and "can I prerender" very soon with @mkhatib 's work. I'm leaning toward a hard and fast rule for now in one place and get the prerendering story done before tuning this deeper.